src/pages/storage/StorageBuckets.tsx

```diff
@@ -0,0 +1,310 @@
+import CreateBucketBtn from "./actions/CreateBucketBtn";
+import SelectableMainTable from "components/SelectableMainTable";
+import SelectedTableNotification from "components/SelectedTableNotification";
+import ResourceLink from "components/ResourceLink";
+import usePanelParams, { panels } from "util/usePanelParams";
+import CreateBucketPanel from "./panels/CreateBucketPanel";
+import StorageBucketBulkDelete from "./actions/StorageBucketBulkDelete";
+import type { LxdStorageBucket } from "types/storage";
+
+const StorageBuckets: FC = () => {
+  const docBaseLink = useDocs();
+  const notify = useNotify();
+  const { project } = useParams<{ project: string }>();
+  const [searchParams] = useSearchParams();
+
+  const [selectedNames, setSelectedNames] = useState<string[]>([]);
+  const [processingNames, setProcessingNames] = useState<string[]>([]);
+  const panelParams = usePanelParams();
+
+  const filters: StorageBucketsFilterType = {
+    queries: searchParams.getAll(QUERY),
+    pools: searchParams.getAll(POOL),
+  };
+
+  if (!project) {
+    return <>Missing project</>;
+  }
+
+  const { data: buckets = [], error, isLoading } = useLoadBuckets(project);
+
+  const getBucketKey = (bucket: LxdStorageBucket) => {
+    return `${bucket.name}-${bucket.pool}-${bucket.location || ""}`;
+  };
+
+  useEffect(() => {
+    const validKeys = new Set(buckets.map(getBucketKey));
+    const validSelections = selectedNames.filter((name) => validKeys.has(name));
+    if (validSelections.length !== selectedNames.length) {
+      setSelectedNames(validSelections);
+    }
+  }, [buckets]);
+
+  if (error) {
+    notify.failure("Loading storage buckets failed", error);
+  }
+
+  const headers = [
+    {
+      content: NAME_COL,
+      sortKey: "name",
+      className: "name",
+    },
+    {
+      content: POOL_COL,
+      sortKey: "pool",
+      className: "pool",
+    },
+    {
+      content: SIZE_COL,
+      sortKey: "size",
+      className: "size",
+    },
+    {
+      content: URL_COL,
+    },
+    {
+      content: "",
+      className: "actions u-align--right",
+      "aria-label": "Actions",
+    },
+  ];
+
+  const filteredBuckets = buckets.filter((item) => {
+    if (!filters.queries.every((q) => item.name.toLowerCase().includes(q))) {
+      return false;
+    }
+    if (filters.pools.length > 0 && !filters.pools.includes(item.pool)) {
+      return false;
+    }
+    return true;
+  });
+
+  const rows = filteredBuckets.map((bucket) => {
+    return {
+      key: getBucketKey(bucket),
+      name: getBucketKey(bucket),
+      className: "u-row",
+      columns: [
+        {
+          content: (
+            <div className="u-truncate" title={bucket.name}>
+              {bucket.name}
+            </div>
+          ),
+          role: "rowheader",
+          "aria-label": NAME_COL,
+        },
+        {
+          content: (
+            <ResourceLink
+              type={"bucket"}
```

Comment:
```suggestion
              type="pool"
```
Sorry for the back and forth, but I think this is a pool and not a bucket.